### ISSUE TITLE
Compilation error on OTP 24-RC2

### ISSUE THREAD

**Issue by LeandroRezendeCoutinho** [asdf-vm/asdf-erlang#193]
Error:
 CXX    x86_64-pc-linux-gnu/wxe_gl.o
gen/wxe_wrapper_8.cpp: In function ‘void wxWindow_GetChildren(WxeApp*, wxeMemEnv*, wxeCommand&)’:
gen/wxe_wrapper_8.cpp:1534:31: error: cannot convert ‘const wxWindowList’ to ‘const wxObjectList&’
 1534 |   rt.send(  rt.make_list_objs(Result, app, "wxWindow"));
      |                               ^~~~~~
      |                               |
      |                               const wxWindowList
In file included from gen/wxe_wrapper_8.cpp:26:
gen/../wxe_return.h:65:53: note:   initializing argument 1 of ‘ERL_NIF_TERM wxeReturn::make_list_objs(const wxObjectList&, WxeApp*, const char*)’
   65 |     ERL_NIF_TERM make_list_objs(const wxObjectList& wx_list, WxeApp *app, const char *cname);
      |                                 ~~~~~~~~~~~~~~~~~~~~^~~~~~~
gen/wxe_wrapper_6.cpp: In function ‘void wxSizer_GetChildren(WxeApp*, wxeMemEnv*, wxeCommand&)’:
gen/wxe_wrapper_6.cpp:1589:31: error: cannot convert ‘const wxSizerItemList’ to ‘const wxObjectList&’
 1589 |   rt.send(  rt.make_list_objs(Result, app, "wxSizerItem"));
      |                               ^~~~~~
      |                               |
      |                               const wxSizerItemList
In file included from gen/wxe_wrapper_6.cpp:26:
gen/../wxe_return.h:65:53: note:   initializing argument 1 of ‘ERL_NIF_TERM wxeReturn::make_list_objs(const wxObjectList&, WxeApp*, const char*)’
   65 |     ERL_NIF_TERM make_list_objs(const wxObjectList& wx_list, WxeApp *app, const char *cname);
      |                                 ~~~~~~~~~~~~~~~~~~~~^~~~~~~
gen/wxe_wrapper_5.cpp: In function ‘void wxMenu_GetMenuItems(WxeApp*, wxeMemEnv*, wxeCommand&)’:
gen/wxe_wrapper_5.cpp:1095:31: error: cannot convert ‘const wxMenuItemList’ to ‘const wxObjectList&’
 1095 |   rt.send(  rt.make_list_objs(Result, app, "wxMenuItem"));
      |                               ^~~~~~
      |                               |
      |                               const wxMenuItemList
In file included from gen/wxe_wrapper_5.cpp:26:
gen/../wxe_return.h:65:53: note:   initializing argument 1 of ‘ERL_NIF_TERM wxeReturn::make_list_objs(const wxObjectList&, WxeApp*, const char*)’
   65 |     ERL_NIF_TERM make_list_objs(const wxObjectList& wx_list, WxeApp *app, const char *cname);
      |                                 ~~~~~~~~~~~~~~~~~~~~^~~~~~~
 CXX    x86_64-pc-linux-gnu/gl_nif.o
gmake[3]: *** [Makefile:163: x86_64-pc-linux-gnu/wxe_wrapper_8.o] Erro 1
gmake[3]: ** Esperando que outros processos terminem.
gmake[3]: *** [Makefile:163: x86_64-pc-linux-gnu/wxe_wrapper_6.o] Erro 1
gmake[3]: *** [Makefile:163: x86_64-pc-linux-gnu/wxe_wrapper_5.o] Erro 1
gmake[3]: Saindo do diretório '/home/leandro/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0-rc2/otp_src_24.0-rc2/lib/wx/c_src'

gmake[2]: *** [/home/leandro/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0-rc2/otp_src_24.0-rc2/make/otp_subdir.mk:29: opt] Erro 2
gmake[2]: Saindo do diretório '/home/leandro/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0-rc2/otp_src_24.0-rc2/lib/wx'

gmake[1]: *** [/home/leandro/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0-rc2/otp_src_24.0-rc2/make/otp_subdir.mk:29: opt] Erro 2
gmake[1]: Saindo do diretório '/home/leandro/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0-rc2/otp_src_24.0-rc2/lib'

gmake: *** [Makefile:498: libs] Erro 2

-------------------------
Info: OS -> OpenSuse.

**Comment by aseigo** [comment]
Ran into this same issue. Installing the 2.8 compat lib packages (`zypper search wx | grep compat`) as noted in the erlang build notes relating to wx was not enough, either. I ended up having to disable all the wx-related build options (`--without-wx --without-debugger --without-observer --without-et`). Seems more an issue with otp24 and their port to wx 3 really being a port to 3.0 and not 3.1 (also pointing, perhaps, to the poor choice that wx is in general here). Anyways, I don't think this is an asdf issue but an upstream problem.
